source/UF/SystemFNotionOfResizing.lagda

```diff
@@ -0,0 +1,348 @@
+--------------------------------------------------------------------------------
+title:        System F Resizing considered as an axiom
+authors:      ["Sam Speight", "Ayberk Tosun"]
+date-started: 2024-05-21
+--------------------------------------------------------------------------------
+
+This module contains some notes from various discussions with Sam Speight.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Subsingletons
+
+module UF.SystemFNotionOfResizing
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+        (pt : propositional-truncations-exist)
+       where
+
+open import InjectiveTypes.Resizing
+open import MLTT.Spartan
+open import UF.Equiv
+open import UF.EquivalenceExamples
+open import UF.Logic
+open import UF.NotNotStablePropositions
+open import UF.Retracts
+open import UF.Size
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+
+open AllCombinators pt fe
+
+\end{code}
+
+One can consider System F resizing in a universe polymorphic form, but we think
+this should be inconsistent due to some form of Girard’s Paradox. This is
+because it gives nested impredicative universes which is known to be
+inconsistent. However, there are lots of details to check here.
+
+\begin{code}
+
+Generalized-System-F-Resizing : (𝓤 𝓥 : Universe) → (𝓤 ⊔ 𝓥) ⁺  ̇
+Generalized-System-F-Resizing 𝓤 𝓥 =
+ (A : 𝓤 ⊔ 𝓥  ̇) → (B : A → 𝓤  ̇) → (Π x ꞉ A , B x) is 𝓤 small
+
+\end{code}
+
+TODO: prove that this generalized form is inconsistent.
+
+
+The special case of this notion of resizing where we pick `𝓤 := 𝓤₀` and
+`𝓥 := 𝓤₁` should be consistent.
+
+\begin{code}
+
+System-F-Resizing : 𝓤₂  ̇
+System-F-Resizing =
+ (A : 𝓤₁  ̇) → (B : A → 𝓤₀  ̇) → (Π x ꞉ A , B x) is 𝓤₀ small
+
+\end{code}
+
+One could also consider the propositional form of this notion of resizing.
+
+\begin{code}
+
+Propositional-System-F-Resizing : 𝓤₂  ̇
+Propositional-System-F-Resizing =
+ (A : 𝓤₁  ̇) →
+  (P : A → Ω 𝓤₀) →
+   (Ɐ x ꞉ A , P x) holds is 𝓤₀ small
+
```

Comment:
If you choose `A` to be a proposition and `P` to be constantly `true`, then `(Ɐ x ꞉ A , P x) holds` is `A x 1` which is equivalent to A. Therefore, combining this with the statement in lines 88-89, we get that `Propositional-System-F-Resizing` is the same thing as propositional resizing.

Reply:
> If you choose `A` to be a proposition and `P` to be constantly `true`, then `(Ɐ x ꞉ A , P x) holds` is `A x 1`

That would be `A ⇒ 𝟙`, which is just `𝟙`, and not `A × 𝟙`.

In fact, this implication would mean that `System-F-Resizing` is equivalent to `propositional-resizing`, which cannot be the case since the former is validated and the latter is refuted in the cubical assemblies model. I learned this from @slspeight.

So I think `Propositional-System-F-Resizing` is actually a sensible thing to consider.